portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -15,18 +15,78 @@
 import javax.servlet.http.HttpServletResponse;
 
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+
+import java.util.List; 
+import java.util.Date;
+import java.text.DateFormat;  
+import java.text.SimpleDateFormat;  
+
+import java.util.ArrayList;
+import java.util.stream.Collectors; 
+import com.google.sps.data.Comment;
+
 /** Servlet that returns some example content. TODO: modify this file to handle comments data */
-@WebServlet("/data")
+@WebServlet("/comments")
 public class DataServlet extends HttpServlet {
 
-  @Override
-  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
-  }
+    ArrayList<Comment> commentList = new ArrayList<>();
+
+    @Override
+    public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+        
+        // Retrieving Comments from DataStore
+        Query query = new Query("Comment").addSort("timestamp", SortDirection.DESCENDING);
+        DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+        PreparedQuery results = datastore.prepare(query);
+
+        // Create ArrayList of all comments retrieved form DataStore
+        List<Comment> commentList = new ArrayList<>();
+        for (Entity entity : results.asIterable()) {
+            String name = (String) entity.getProperty("name");
+            String message = (String) entity.getProperty("message");
+            long timestamp = (long) entity.getProperty("timestamp"); 
+            long id = entity.getKey().getId();
+
+            Comment newComment = new Comment(name, message, timestamp, id);
+            commentList.add(newComment);
+        }
+
+        // Convert commentList to JSON and send response
+        Gson gson = new Gson();
+        response.setContentType("application/json;");
+        response.getWriter().println(gson.toJson(commentList));
+    }
+
+    @Override
+    public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+        // To be used for timestamp
+        long timestamp = System.currentTimeMillis();
+
+        // Create JSON object with GSON
+        String requestData = request.getReader().lines().collect(Collectors.joining());
+        Comment comment = new Gson().fromJson(requestData, Comment.class);
+
+        // Creating DataStore Entity
+        Entity taskEntity = new Entity("Comment");
+        taskEntity.setProperty("name", comment.getName());
+        taskEntity.setProperty("message", comment.getMessage());
+        taskEntity.setProperty("timestamp", timestamp);
```

Comment:
create constants for these strings so we know there won't be typo when getting/setting values.